src/main/java/de/tum/in/www1/artemis/service/connectors/localci/scaparser/ReportParser.java

```diff
@@ -61,11 +65,29 @@ public StaticCodeAnalysisReportDTO transformToReport(File file) {
                 return createFileTooLargeReport(file.getName());
             }
 
-            ParserContext context = new ParserContext();
-            return context.getReport(file);
+            return getReport(file);
         }
         catch (Exception e) {
             return createErrorReport(file.getName(), e);
         }
     }
+
+    /**
+     * Builds the document using the provided file and parses it to a Report object using ObjectMapper.
+     *
+     * @param file File referencing the static code analysis report
+     * @return Report containing the static code analysis issues
+     * @throws UnsupportedToolException if the static code analysis tool which created the report is not supported
+     * @throws IOException              if the file could not be read
+     */
+    public static StaticCodeAnalysisReportDTO getReport(File file) throws IOException {
+        String xmlContent = Files.readString(file.toPath());
+        return getReport(xmlContent, file.getName());
+    }
+
+    public static StaticCodeAnalysisReportDTO getReport(String xmlContent, String fileName) {
+        ParserPolicy parserPolicy = new ParserPolicy();
+        ParserStrategy parserStrategy = parserPolicy.configure(fileName);
+        return parserStrategy.parse(xmlContent);  // Pass the whole XML content directly
```

Comment:
```suggestion
        return parserStrategy.parse(xmlContent);  // Parse the whole XML content directly
```
Typo or on purpose?

Reply:
This was on purpose (pass the content string to the method), but I don't think the comment is really necessary.